src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2230,14 +2230,15 @@ public override BoundNode VisitConditionalAccess(BoundConditionalAccess node)
 
             var accessExpressionType = VisitRvalueWithState(node.AccessExpression);
             Join(ref this.State, ref receiverState);
-            // Per LDM 2019-02-13 decision, the result of a conditional access might be null even if
-            // both the receiver and right-hand-side are believed not to be null.
-            NullableFlowState resultState = NullableFlowState.MaybeNull;
 
             // https://github.com/dotnet/roslyn/issues/29956 Use flow analysis type rather than node.Type
             // so that nested nullability is inferred from flow analysis. See VisitConditionalOperator.
             TypeSymbol type = node.Type;
 
+            // Per LDM 2019-02-13 decision, the result of a conditional access might be null even if
+            // both the receiver and right-hand-side are believed not to be null.
+            NullableFlowState resultState = type.CanContainNull() ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
```

Comment:
>type.CanContainNull() ? NullableFlowState.MaybeNull : NullableFlowState.NotNull [](start = 44, length = 79)

`NullableFlowState.MaybeNull`, since the normalization is done in `TypeWithState.Create`